### DISCUSSION TITLE
How to toggle the theme based on system UI automatically?

### DISCUSSION THREAD

**Discussion by martin-braun** [NvChad/NvChad#1600] [Q&A] [answered]
My way to handle this situation is using `f-person/auto-dark-mode.nvim` like so: 

```lua
local light_theme = "one_light"
local dark_theme = "onedark"
M.ui.theme_toggle = { light_theme, dark_theme }
M.ui.theme = light_theme

M.plugins = {
    ...
    ["f-person/auto-dark-mode.nvim"] = {
        config = function ()
            require('auto-dark-mode').setup({
                update_interval = 1000,
                set_light_mode = function()
                    if vim.g.nvchad_theme ~= light_theme then
                        require("base46").toggle_theme()
                        print("switched to " .. vim.g.nvchad_theme .. " theme!")
                    end
                end,
                set_dark_mode = function()
                    if vim.g.nvchad_theme ~= dark_theme then
                        require("base46").toggle_theme()
                        print("switched to " .. vim.g.nvchad_theme .. " theme!")
                    end
                end,
            })
            require('auto-dark-mode').init()
        end,
    },
    ...
}
```

In theory this should work, but `toggle_theme` will search-replace all theme strings in chadrc.lua which messes up my logic. Now I have to encode my theme names or is there a way to disable persistent theme toggling?

**Comment by martin-braun** [comment, marked as answer]
Okay I dug and found the [code](https://github.com/NvChad/extensions/blob/main/lua/nvchad/change_theme.lua#L25) responsible for the search-replace in `chadrc.lua`.

I can avoid this issue by simply renaming my variables to not end with "theme". 

I don't like the fact that this behavior is not well documented, maybe you can leave a note somewhere in the docs, @siduck?

**Reply by martin-braun** [reply]
My result: 

```lua
local M = { options = {}, ui = {}, plugins = {}, mappings = {} }

M.ui.theme_toggle = { "one_light", "onedark" }
M.ui.theme = "onedark"

M.plugins = {
    ...
    ["f-person/auto-dark-mode.nvim"] = {
        config = function ()
            vim.g.toggle_theme_icon = ""
            local config = require("core.utils").load_config()
            require('auto-dark-mode').setup({
                update_interval = 1000,
                set_light_mode = function()
                    if config.ui.theme ~= config.ui.theme_toggle[1] then
                        require("base46").toggle_theme()
                        vim.g.toggle_theme_icon = ""
                        config.ui.theme = vim.g.nvchad_theme
                        print("switched to " .. vim.g.nvchad_theme .. " theme!")
                    end
                end,
                set_dark_mode = function()
                    if config.ui.theme ~= config.ui.theme_toggle[2] then
                        require("base46").toggle_theme()
                        vim.g.toggle_theme_icon = ""
                        config.ui.theme = vim.g.nvchad_theme
                        print("switched to " .. vim.g.nvchad_theme .. " theme!")
                    end
                end,
            })
            require('auto-dark-mode').init()
        end,
    },
    ...
}

...

return M
```

First theme needs to be the light theme. Auto toggling the theme for MacOS, success.

**Reply by siduck** [reply]
@martin-braun This is the first time someone's asked about that code, its not a thing to be documented!

**Reply by martin-braun** [reply]
@siduck and that's fine. Nobody expects a perfect solution out of the box. My suggestion is to document it though, because it caused some trouble on my end. Things like that are too magical in my opinion, maybe it was better NvChad should store the current theme in the cache and not overwrite my files. If I had my configs in a repo, changing the theme would cause working changes. At the very least it would be good to have this behavior documented. 

Whatever you do out of this, please allow me to always retrieve the current theme, so I can prevent toggling the theme, if I have the right one. Another way to solve this would be to have a function that allows me to toggle to a specific theme on the fly, but base46 isn't exposing any of that as of now.

Thank you.
